File: CS/MeffedNavigationService.Infrastructure/Contract/IModuleListing.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.ComponentModel.Composition;
using MeffedNavigationService.Infrastructure.DataModel;

namespace MeffedNavigationService.Infrastructure.Contract {
    [InheritedExport(typeof(IModuleListing))]
    public interface IModuleListing {
        IEnumerable<MenuDataItem> MenuItems { get; }
    }
}

File: VB/MeffedNavigationService.Infrastructure/Contract/IModuleListing.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.ComponentModel.Composition
Imports MeffedNavigationService.Infrastructure.DataModel

Namespace MeffedNavigationService.Infrastructure.Contract
    <InheritedExport(GetType(IModuleListing))> _
    Public Interface IModuleListing
        ReadOnly Property MenuItems() As IEnumerable(Of MenuDataItem)
    End Interface
End Namespace
